adoc/chapters/device_compiler.adoc

```diff
@@ -437,6 +446,16 @@ the entire device image when [code]#KernelA# is submitted to device
 [code]#dev1#.
 ====
 
+[NOTE]
+====
+As stated in <<sec:language.restrictions.kernels>>, the restrictions
+for optional kernel features do not apply to manifestly
+constant-evaluated expressions or conversions in device code. Device
+code may use optional features in manifestly constant-evaluated
+expressions or conversions even if the device does not support the
+optional feature.
+====
```

Comment:
I wonder if this should be normative text instead of a non-normative note.

Reply:
Done.